contrib/automation_tests/test_cases/symbols_tab.py

```diff
@@ -18,21 +23,229 @@
 from test_cases.capture_window import Capture
 from test_cases.live_tab import VerifyFunctionCallCount
 
+Module = namedtuple("Module", ["name", "path", "is_loaded"])
+CACHE_LOCATION = "{appdata}\\OrbitProfiler\\cache\\".format(appdata=os.getenv('APPDATA'))
+
 
 def _show_symbols_and_functions_tabs(top_window):
     logging.info("Showing symbols tab")
     find_control(top_window, "TabItem", "Symbols").click_input()
 
 
-class LoadSymbols(E2ETestCase):
+def _find_and_close_error_dialog(top_window) -> str or None:
+    window = find_control(top_window, 'Window', 'Error loading symbols', raise_on_failure=False)
+    if window is not None:
+        error_message = find_control(window, 'Text').texts()[0]
+        re_result = re.search('for module "(.+?)"', error_message)
+        if not re_result:
+            return None
+        logging.info("Found error dialog with message {message}, closing.".format(message=error_message))
+        find_control(window, 'Button').click_input()
+        return re_result[1]
+    else:
+        return None
+
+
+class DurationDiffMixin(ABC):
+    """
+    Provides utility methods to store and compare durations between runs, and fails the test if a certain threshold
+    of duration difference is exceeded between runs.
+
+    For example, this is used to measure symbol loading times from cold and warm caches, and verify that warm cache
+    loading is significantly faster.
+
+    Usage:
+    - Inherit your E2ETestCase from this mixin
+    - Use time.time() to record start times and duration
+    - Where needed, call `self._check_and_update_duration` to store the current duration and fail the test if applicable
+    """
+    suite = None
+
+    @abstractmethod
+    def expect_true(self, cond, msg):
+        pass
+
+    @abstractmethod
+    def expect_eq(self, left, right, msg):
+        pass
+
+    def _check_and_update_duration(self, storage_key: str, total_duration: float, expected_difference: float):
```

Comment:
nit: `total_duration` should be `current_duration` ?

Reply:
Yes, thanks for spotting, that's a leftover from code refactoring